openrewrite/src/javascript/parserUtils.ts

```diff
@@ -1,4 +1,67 @@
 import * as ts from "typescript";
+import * as J from '../java'
+import * as JS from "./tree";
+
+export const statement_implementations_list: Array<{ new (...args: any[]): J.Statement }> = [
+    J.Assert,
+    J.Assignment,
+    J.AssignmentOperation,
+    J.Block,
+    J.Break,
+    J.Case,
+    J.ClassDeclaration,
+    J.Continue,
+    J.Empty,
+    J.EnumValueSet,
+    J.Erroneous,
+    J.FieldAccess,
+    J.If,
+    J.Import,
+    J.Label,
+    J.Lambda,
+    J.MethodDeclaration,
+    J.MethodInvocation,
```

Comment:
Do you also need something like an is_expression or how do we make sure that the types which are both expression and statement don't get wrapped? Also note the inverse wrapper (StatementExpression) which might show the same behavior.

Reply:
LGTM. I think it would be best if we could have this generated instead of maintaining it by hand, as now every time we add a new type to either `J` or `JS`, we need to make sure to check if we need to update these arrays.

Reply:
I am missing the following four types from `J`: `DoWhileLoop`, `ForEachLoop`, `ForLoop`, and `WhileLoop`. AFAIK we use all of these except for `ForEachLoop` in our parser.

Reply:
I am also missing the `JS` types `ForOfLoop` and `ForInLoop`.

Reply:
Yes, thanks!
I've used a small util to generate the list of classes, but missed the `Loop` inheritors.

Reply:
> LGTM. I think it would be best if we could have this generated instead of maintaining it by hand, as now every time we add a new type to either `J` or `JS`, we need to make sure to check if we need to update these arrays.

Yes, it will be a more consistent way to keep lists up-to-date.